core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2003,4 +2008,48 @@ public void multipleDiffsAgainstMultipleTablesLastFails() {
     assertThat(schema2.findField("new-column")).isNull();
     assertThat(schema2.columns()).hasSize(1);
   }
+
+  @Test
+  public void testScanReportingOnSerializableTable() throws IOException, ClassNotFoundException {
+    Table table = catalog().buildTable(TABLE, SCHEMA).create();
+    table
+        .newFastAppend()
+        .appendFile(
+            DataFiles.builder(PartitionSpec.unpartitioned())
+                .withPath("/path/to/data-a.parquet")
+                .withFileSizeInBytes(10)
+                .withRecordCount(2)
+                .build())
+        .commit();
+
+    table = catalog().loadTable(TABLE);
+    Mockito.reset(adaptor);
+
+    try (CloseableIterable<FileScanTask> tasks = table.newScan().planFiles()) {
+      assertThat(tasks.iterator()).hasNext();
+    }
+
+    Table serializableTable =
+        TestHelpers.KryoHelpers.roundTripSerialize(SerializableTable.copyOf(table));
+    try (CloseableIterable<FileScanTask> tasks = serializableTable.newScan().planFiles()) {
+      assertThat(tasks.iterator()).hasNext();
+    }
+
+    serializableTable = TestHelpers.roundTripSerialize(SerializableTable.copyOf(table));
+    try (CloseableIterable<FileScanTask> tasks = serializableTable.newScan().planFiles()) {
+      assertThat(tasks.iterator()).hasNext();
+    }
+
+    // once the table is serialized, it should still be able to send metrics reports via REST
+    ResourcePaths paths = ResourcePaths.forCatalogProperties(Maps.newHashMap());
+    verify(adaptor, times(3))
```

Comment:
Why do you need to keep the adapter around? Other tests create their own adapter for assertions like this. For example, `testCatalogBasicBearerToken`. I think you could just create a new one like that.

Reply:
fair point, we don't have to keep the adapter around. I've made this local to the test itself